### PR TITLE
Add gas checker.

### PR DESCRIPTION
**Pull request by chriseth**
Closes https://github.com/ethereum/browser-solidity/issues/162

This includes a refactoring of the error reporting mechanism (add options as an object), adds a feature to it (use "span" instead of "pre") and refactors the static analyzer in response to different requirements, as expected.

### CODE REVIEW

**Comment by axic** [comment]
This has a lot of non-gaschecker code and is missing the actual gaschecker file.

**Comment by chriseth** [comment]
@axic thanks, added the gas checker.

File: src/app/renderer.js [hunk 1]
```diff
@@ -34,13 +34,14 @@ function Renderer (editor, updateFiles, udapp, executionContext, formalVerificat
...
   var self = this
-  if (!type) {
-    type = utils.errortype(message)
+  var opt = options || {}
+  if (!opt.type) {
+    opt.type = utils.errortype(message)
   }
-  var $pre = $('<pre />').text(message)
-  var $error = $('<div class="sol ' + type + '"><div class="close"><i class="fa fa-close"></i></div></div>').prepend($pre)
+  var $pre = $(opt.useSpan ? '<span />' : '<pre />').text(message)
```
Comment (yann300) [inline src/app/renderer.js]: why using span instead of pre ?

Comment (chriseth) [inline src/app/renderer.js]: The pre was mostly needed to format the pointer to the source location properly, which is generated by the compiler. If we use `span` here, the browser breaks the lines and it is much easier to read.

Comment (yann300) [inline src/app/renderer.js]: cause if you put `\n` in it and use a span. it will not add a new line

Comment (chriseth) [inline src/app/renderer.js]: Yes, I noticed that, too. But then you do want the browser to break. Perhaps we can go with html and use `<br/>` for newlines?

Comment (yann300) [inline src/app/renderer.js]: yes

File: test-browser/tests/staticanalysis.js [hunk 2]
```diff
@@ -28,33 +29,22 @@ module.exports = {
...
-      })
-    })
-  browser.assert.equal(found, true)
-}
-
 function runTests (browser) {
   browser
     .waitForElementVisible('.newFile', 10000)
   contractHelper.testContracts(browser, sources.sources.Untitled, ['TooMuchGas', 'test1', 'test2'], function () {
+    console.log('pppzzz?')
```
Comment (chriseth) [inline test-browser/tests/staticanalysis.js]: Please remove.

**Pull request merged**